[Compute] Vm health field in vm instance view

madewithsmiles opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

msJinLei reviewed (changes requested):
RecoveryServices.Backup.Test.ScenarioTests which replies on compute 38.0 fail due to the SDK update. You need to fix the test failure in this PR.

madewithsmiles commented:
@msJinLei can you please confirm if the change was effective and merge in such case? Thank you

grizzlytheodore commented:
there are 6 csproj files where SDK versions need to be updated, I think I only see three in yours. 
https://microsoft.sharepoint.com/teams/AzureComputeSupport/_layouts/OneNote.aspx?id=%2Fteams%2FAzureComputeSupport%2FShared%20Documents%2FNotebooks%2FCompute%20Supportability&wd=target%28SDK_Swagger_Powershell.one%7C9DCABB06-CAB9-40C6-BE56-F84BB2493F9E%2FDevelopment%7C06F21056-5183-47E4-B1E8-ADF1165186C5%2F%29
onenote:https://microsoft.sharepoint.com/teams/AzureComputeSupport/Shared%20Documents/Notebooks/Compute%20Supportability/SDK_Swagger_Powershell.one#Development&section-id={9DCABB06-CAB9-40C6-BE56-F84BB2493F9E}&page-id={06F21056-5183-47E4-B1E8-ADF1165186C5}&end
will show all the places that need to have sdk package reference updated @madewithsmiles

madewithsmiles commented:
Interesting. Will update now. Thanks Theo

madewithsmiles commented:
I have updated all the files. @msJinLei can you please merge when you can?

msJinLei commented:
@madewithsmiles Please fix the credential scan error https://dev.azure.com/azure-sdk/internal/_build/results?buildId=504760&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=a4822bc1-f3a9-5984-c941-6c1b7aa2ab81

Sandido commented on src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:
```diff
@@ -2994,6 +2995,66 @@ function Test-VirtualMachineGetStatus
 
+<#
+.SYNOPSIS
+Test Virtual Machines
```
Improve this description. It does not describe what the test does.

madewithsmiles replied:
Improved. As a note, my description is the only one in the virtualmachinetests file. This means that we need to decide as a team a directive for descriptions since no one has been doing it before. Resolving.

Sandido commented on src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:
```diff
@@ -2994,6 +2995,66 @@ function Test-VirtualMachineGetStatus
+    try
+    {
+        # Common
+        if ($loc -eq $null)
```
$loc will always be null here. Just use the assignment line below this instead of having an if check.

madewithsmiles replied:
True! Resolving.

Sandido commented on src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:
```diff
@@ -2994,6 +2995,66 @@ function Test-VirtualMachineGetStatus
+        $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+        # Check for VmHealth Property
+        Assert-NotNull $vm.VMHealth
```
Instead of just checking they are not null, can you check for specific values they should be? That would be a more helpful test.

madewithsmiles replied:
The values might change....I think it's due to the deployment of the extension and the status of the vm when the extension is deployed so checking for a specific value would be cumbersome. Anyways, the most important idea for this test is that VMHealth **_is not always returned_** in the properties of the instance view; it is returned only _if there is a health extension_.  VMHealth has some values but the message might be different. This test captures the intended behavior. For your reference, you can read the intended change here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/667. You can also review my demo in the folders in teams. Resolving

Sandido commented on src/Compute/Compute/ChangeLog.md:
```diff
@@ -19,6 +19,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Added `vmHealh` property to ``Get-AzVm ... -Status``'s return object (VMInstanceView)
```
VmHealth

Sandido replied:
Perhaps rephrase to
'Added 'VmHealth' property to the return object from ``Get-AzVm`` when the parameter ``-Status`` is used.

madewithsmiles replied:
Resolving

msJinLei commented:
/azp run

madewithsmiles commented:
@Sandido I cannot push my changes without your approval. Can you please approve my changes? I addressed all your comments, and @grizzlytheodore reviewed as well.

Sandido commented on src/Compute/Compute/ChangeLog.md:
```diff
@@ -19,7 +19,7 @@
 -->
 ## Upcoming Release
-* Added `vmHealh` property to ``Get-AzVm ... -Status``'s return object (VMInstanceView)
+* Added ``vmHealh`` property to the virtual machine's instance view, which is the returned object when ``Get-AzVm`` is invoked with ``-Status``
```
Please correct the spelling to VmHealth as I commented before.

madewithsmiles replied:
Just checked. The field name is VMHealth so I am changing to that.

Sandido commented:
@msJinLei , please review.

msJinLei commented:
/azp run
